### PR TITLE
(feat, java): add timeout to request options

### PR DESCRIPTION
**Pull request by armandobelardo**
Addresses https://github.com/fern-api/fern/issues/2922 for Java

### CODE REVIEW

File: generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java [hunk 1]
```diff
@@ -25,14 +25,9 @@
...
 import com.fern.java.output.GeneratedJavaFile;
-import com.squareup.javapoet.ClassName;
-import com.squareup.javapoet.CodeBlock;
-import com.squareup.javapoet.FieldSpec;
-import com.squareup.javapoet.JavaFile;
-import com.squareup.javapoet.MethodSpec;
-import com.squareup.javapoet.ParameterSpec;
-import com.squareup.javapoet.ParameterizedTypeName;
-import com.squareup.javapoet.TypeSpec;
+import com.squareup.javapoet.*;
```
Comment (dsinghvi) [inline generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java:28]: we'll need to fix imports

File: generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java [hunk 2]
```diff
@@ -51,6 +46,11 @@ public final class RequestOptionsGenerator extends AbstractFileGenerator {
                     Modifier.FINAL)
             .build();
 
+    private static final FieldSpec.Builder TIMEOUT_FIELD_BUILDER = FieldSpec.builder(
+                    ParameterizedTypeName.get(ClassName.get(Optional.class), TypeName.get(Integer.class)),
+                    "timeout",
+                    Modifier.PRIVATE);
```
Comment (dsinghvi) [inline generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java:52]: in java i think its common to do `timeout(10, TimeUnit.Seconds)` -- thats what okhttp does

File: generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java (hunk 1 above)
Comment (armandobelardo) [reply generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java:28]: Just fixed my intellij settings to not do this

File: generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java (hunk 2 above)
Comment (armandobelardo) [reply generators/java/sdk/src/main/java/com/fern/java/client/generators/RequestOptionsGenerator.java:52]: Will do! I was mirroring some of the others which is just in seconds

File: seed/java-sdk/bytes/src/main/java/com/seed/bytes/core/RequestOptions.java [hunk 3]
```diff
@@ -19,8 +36,22 @@ public static Builder builder() {
...
+
+        public Builder timeout(Optional<Integer> timeout) {
+            this.timeout = timeout;
+            return this;
+        }
+
+        public Builder timeoutTimeUnit(TimeUnit timeoutTimeUnit) {
+            this.timeoutTimeUnit = timeoutTimeUnit;
+            return this;
+        }
```
Comment (dsinghvi) [inline seed/java-sdk/bytes/src/main/java/com/seed/bytes/core/RequestOptions.java:51]: `timeout(5, TimeUnit.Seconds)`

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**